### PR TITLE
Feature/get space by name

### PR DESCRIPTION
**Pull request by roman-kozachenko**

### CODE REVIEW

File: src/tarantool.client/Box.cs [hunk 1]
```diff
@@ -35,10 +41,35 @@ public Box(ConnectionOptions options)
...
+        public async Task<DataResponse<TResponse[]>> Eval<TTuple, TResponse>(string expression, TTuple request)
+           where TTuple : ITuple
+        {
+            var evalRequest = new EvalRequest<TTuple>(expression, request);
+            return await _logicalConnection.SendRequest<EvalRequest<TTuple>, TResponse>(evalRequest);
         }
 
-        private async Task TryLogin(GreetingsResponse greetings)
+        private async Task LoginIfNotGuest(GreetingsResponse greetings)
         {
```
Comment (aensidhe) [inline src/tarantool.client/Box.cs]: We need to provide more meaningful message. A rule of thumb:
- What's happening
- How to fix

File: src/tarantool.client/Converters/ResponseHeaderConverter.cs [hunk 2]
```diff
@@ -30,18 +31,31 @@ public void Write(ResponseHeader value, IMsgPackWriter writer)
...
         {
-            reader.ReadMapLength().ShouldBe(3u);
+            var length = reader.ReadMapLength();
+            if (length != 3u)
+            {
+                throw ExceptionHelper.InvalidMapLength(3u, length);
+            }
 
-            _keyConverter.Read(reader).ShouldBe(Key.Code);
+            ReadKey(reader, Key.Code);
```
Comment (aensidhe) [inline src/tarantool.client/Converters/ResponseHeaderConverter.cs]: I will ask: will tarantool guarantee order of fields. But in general case it is not very to rely on order.

File: src/tarantool.client/LogicalConnection.cs [hunk 3]
```diff
@@ -84,10 +64,50 @@ public IEnumerable<TaskCompletionSource<MemoryStream>> PopAllResponseCompletionS
...
+            var requestId = GetRequestId();
+            var responseTask = GetResponseTask(requestId);
+
+            long headerLength;
+            var headerBuffer = CreateAndSerializeBuffer(request, requestId, serializedRequest, out headerLength);
+
+            await _physicalConnection.Write(headerBuffer, 0, Constants.PacketSizeBufferSize + (int)headerLength);
+            await _physicalConnection.Write(serializedRequest, 0, serializedRequest.Length);
+
+            var responseBytes = await responseTask;
```
Comment (aensidhe) [inline src/tarantool.client/LogicalConnection.cs]: Right here we can look and see the beauty of tasks/promises.

File: src/tarantool.client/LogicalConnection.cs [hunk 4]
```diff
@@ -84,10 +64,50 @@ public IEnumerable<TaskCompletionSource<MemoryStream>> PopAllResponseCompletionS
...
+            stream.Seek(0, SeekOrigin.Begin);
+            MsgPackSerializer.Serialize(packetLength, stream, _msgPackContext);
+            return packetSizeBuffer;
+        }
+
         private RequestId GetRequestId()
         {
             var ulongRequestId = (long)_currentRequestId.Value;
-            Interlocked.CompareExchange(ref ulongRequestId, 0, MaxRequestId);
+            Interlocked.CompareExchange(ref ulongRequestId, 0, Constants.MaxRequestId);
```
Comment (aensidhe) [inline src/tarantool.client/LogicalConnection.cs]: Actually, we don't need this. Increment don't throw exceptions.

File: src/tarantool.client/Box.cs (hunk 1 above)
Comment (roman-kozachenko) [inline src/tarantool.client/Box.cs]: Fixed.
Next time please comment line with exception throwing instruction.

File: src/tarantool.client/Converters/ResponseHeaderConverter.cs (hunk 2 above)
Comment (roman-kozachenko) [inline src/tarantool.client/Converters/ResponseHeaderConverter.cs]: Totally  agree with you.

File: src/tarantool.client/LogicalConnection.cs (hunk 4 above)
Comment (roman-kozachenko) [inline src/tarantool.client/LogicalConnection.cs]: Great catch, I had to check that.

**Pull request merged**